packages/amplify-cli/src/execution-manager.ts

```diff
@@ -310,6 +310,16 @@ const raisePostPullEvent = async (context: Context): Promise<void> => {
 };
 
 const raisePostCodegenModelsEvent = async (context: Context): Promise<void> => {
+  const optionsIndicatingUninitializedModelgen = ['target', 'model-schema'];
+  const cliOptions = context?.parameters?.options ? new Set(Object.keys(context.parameters.options)) : new Set();
+  if (optionsIndicatingUninitializedModelgen.every((option) => cliOptions.has(option))) {
+    console.log(
```

Comment:
nit: `printer.info`

Reply:
Updating to use the printer, and also adding to preCodegenModels, per offline discussion.